lib/node_modules/@stdlib/math/base/special/cosd/src/main.c

```diff
@@ -34,12 +37,30 @@
 */
 double stdlib_base_cosd( const double x ) {
 	double xRad;
-	if ( stdlib_base_is_infinite( x ) ) {
+	double rx;
+
+	if ( stdlib_base_is_infinite( x ) || stdlib_base_is_nan( x ) ) {
 		return 0.0 / 0.0; // NaN
 	}
-	if ( stdlib_base_is_integer( ( ( x / 90.0 ) - 1.0 ) / 2.0 ) ) {
-		return 0.0;
+
+	rx = stdlib_base_abs( stdlib_base_fmod( x, 360.0 ) );
+
+	if ( rx <= 45.0 ) {
+		xRad = stdlib_base_deg2rad( rx );
+		return stdlib_base_kernel_cos( xRad, 0.0 );
```

Comment:
```suggestion
		return stdlib_base_kernel_cos( stdlib_base_deg2rad( rx ), 0.0 );
```
@anandkaranubc Can we directly do something like this, eliminating the need for `xRad`?

Reply:
Sure! Doing it right now. The only reason I did it that way was to improve code readability and keep the code structured.

Reply:
I would say we typically try to avoid temporary variables which can be readily inlined, so long as the expressions are relatively simple. In this case, Gunj's suggestion is generally appropriate.

Reply:
Well noted!  

Thanks, @gunjjoshi, for the suggestion. 🙌